### PR TITLE
doc-gen: migrate scalar functions (crypto) documentation

### PR DESCRIPTION
**Pull request by Chen-Yuan-Lai**
## Which issue does this PR close?

part of #13671.

## What changes are included in this PR?

As discussed in #13671, this PR will migrate the builtin hashing functions documentation that currently support migration.

### CODE REVIEW

**Comment by goldmedal** [comment]
Thanks @Chen-Yuan-Lai for working on this
It seems the CI failure can fixed through the following command
```
- Run # If you encounter an error, run './dev/update_function_docs.sh' and commit
  # If you encounter an error, run './dev/update_function_docs.sh' and commit
  ./dev/update_function_docs.sh
```
Same as other related PRs.

File: datafusion/functions/src/crypto/digest.rs [hunk 1]
```diff
@@ -19,13 +19,38 @@
...
+        name = "algorithm",
+        description = "String expression specifying algorithm to use. Must be one of:       
+- md5
+- sha224
+- sha256
+- sha384
+- sha512
+- blake2s
+- blake2b
+- blake3"
```
Comment (goldmedal) [inline datafusion/functions/src/crypto/digest.rs:51]:
```suggestion
    - md5
    - sha224
    - sha256
    - sha384
    - sha512
    - blake2s
    - blake2b
    - blake3"
```
Maybe we can add some indents for them to format them like
Screenshot 2024-12-28 at 11 38 10 AM

**Review by Omega359 — APPROVED** [review]
lgtm

**Review by goldmedal — APPROVED** [review]

**Comment by goldmedal** [comment]
Thanks @Chen-Yuan-Lai and @Omega359 for the reviews.

By the way, I modified the description `close #13671` to `part of #13671` because I think there are other remaining PR waiting for reviewing. We shouldn't close #13671 by this PR.

**Pull request merged**